native/arrow_format_nif/src/message/schema/field.rs

```diff
@@ -0,0 +1,42 @@
+use crate::utils::CustomMetadata;
+use rustler::{Atom, Decoder, Encoder, NifRecord, Term};
+
+#[derive(Debug, NifRecord)]
+#[tag = "field"]
+pub struct Field {
+    pub name: Name,
+    pub nullable: bool,
+    pub r#type: Atom,
+    pub dictionary: Atom,
+    pub children: Vec<Field>,
+    pub custom_metadata: CustomMetadata,
+}
+
+// TODO: Update Option<T>'s Encoder and Decoder implementations to treat
+// undefined as None in rustler main.
```

Comment:
I didn't understand this comment

Reply:
This comment was from before I was aware of `ErlOption`. In Erlang, the convention is to use `undefined` instead of `nil`, so `Option` doesn't work. This comment was just a reminder for me to write something for ErlOption to eliminate the Name Enum.

Then I realised that rustler has poor charlist support.

I will delete this comment.

Reply:
> Then I realised that rustler has poor charlist support.

It may be worth moving to binaries for strings anyway. Erlang projects such as Hackney and Cowboy already do that.

Reply:
I probably should. But sometime later.